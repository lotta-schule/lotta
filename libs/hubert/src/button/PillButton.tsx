import { Button, ButtonProps } from './Button';
import clsx from 'clsx';

import styles from './PillButton.module.scss';

export type PillButtonProps = Omit<ButtonProps, 'onlyIcon'>;

export const PillButton = ({ className, ...props }: PillButtonProps) => {
  return (
    <Button
      onlyIcon
      className={clsx(className, styles.root, styles.button)}
      {...props}
    />
  );
};
