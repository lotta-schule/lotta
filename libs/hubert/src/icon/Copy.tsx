import * as React from 'react';
import clsx from 'clsx';

import styles from './SvgIcon.module.scss';

export const Copy = React.memo(
  React.forwardRef<SVGSVGElement, React.SVGProps<SVGSVGElement>>(
    ({ className, ...props }, ref) => (
      <svg
        xmlns="http://www.w3.org/2000/svg"
        focusable="false"
        viewBox="0 -960 960 960"
        aria-hidden="true"
        className={clsx(styles.root, className)}
        data-testid="copy-icon"
        ref={ref}
        {...props}
      >
        <path d="M360-240q-33 0-56.5-23.5T280-320v-480q0-33 23.5-56.5T360-880h360q33 0 56.5 23.5T800-800v480q0 33-23.5 56.5T720-240H360Zm0-80h360v-480H360v480ZM200-80q-33 0-56.5-23.5T120-160v-560h80v560h440v80H200Zm160-240v-480 480Z" />
      </svg>
    )
  )
);
Copy.displayName = 'CopyIcon';
