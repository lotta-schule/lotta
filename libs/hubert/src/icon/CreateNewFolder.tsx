import * as React from 'react';
import clsx from 'clsx';

import styles from './SvgIcon.module.scss';

export const CreateNewFolder = React.memo(
  ({ className, ...props }: React.SVGProps<SVGSVGElement>) => (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      focusable="false"
      viewBox="0 0 24 24"
      aria-hidden="true"
      className={clsx(styles.root, className)}
      data-testid="create-new-folder-icon"
      {...props}
    >
      <path d="M20 6h-8l-2-2H4c-1.11 0-1.99.89-1.99 2L2 18c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2m-1 8h-3v3h-2v-3h-3v-2h3V9h2v3h3z" />
    </svg>
  )
);
CreateNewFolder.displayName = 'CreateNewFolderIcon';
