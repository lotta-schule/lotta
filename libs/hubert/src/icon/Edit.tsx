import * as React from 'react';
import clsx from 'clsx';

import styles from './SvgIcon.module.scss';

export const Edit = React.memo(
  ({ className, ...props }: React.SVGProps<SVGSVGElement>) => (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      focusable="false"
      viewBox="0 0 24 24"
      aria-hidden="true"
      className={clsx(styles.root, className)}
      data-testid="edit-icon"
      {...props}
    >
      <path d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 0 0-1.41 0l-1.83 1.83 3.75 3.75z" />
    </svg>
  )
);
Edit.displayName = 'EditIcon';
