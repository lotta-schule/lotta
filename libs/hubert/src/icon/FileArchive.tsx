import * as React from 'react';
import clsx from 'clsx';

import styles from './SvgIcon.module.scss';

export const FileArchive = React.memo(
  ({ className, ...props }: React.SVGProps<SVGSVGElement>) => (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      focusable="false"
      viewBox="0 0 384 512"
      aria-hidden="true"
      className={clsx(styles.root, className)}
      data-testid="file-archive-icon"
      {...props}
    >
      <path
        fill="#ADADAD"
        d="M64,0C28.7,0,0,28.7,0,64v384c0,35.3,28.7,64,64,64h256c35.3,0,64-28.7,64-64V160H256c-17.7,0-32-14.3-32-32V0
	H64z M256,0v128h128L256,0z M96,48c0-8.8,7.2-16,16-16h32c8.8,0,16,7.2,16,16s-7.2,16-16,16h-32C103.2,64,96,56.8,96,48z M96,112
	c0-8.8,7.2-16,16-16h32c8.8,0,16,7.2,16,16s-7.2,16-16,16h-32C103.2,128,96,120.8,96,112z M96,176c0-8.8,7.2-16,16-16h32
	c8.8,0,16,7.2,16,16s-7.2,16-16,16h-32C103.2,192,96,184.8,96,176z M89.7,247.8c3.7-14,16.4-23.8,30.9-23.8h14.8
	c14.5,0,27.2,9.7,30.9,23.8l23.5,88.2c1.4,5.4,2.1,10.9,2.1,16.4c0,35.2-28.8,63.7-64,63.7s-64-28.5-64-63.7
	c0-5.5,0.7-11.1,2.1-16.4l23.5-88.2H89.7z M112,336c-8.8,0-16,7.2-16,16s7.2,16,16,16h32c8.8,0,16-7.2,16-16s-7.2-16-16-16H112z"
      />
    </svg>
  )
);
FileArchive.displayName = 'FileArchiveIcon';
