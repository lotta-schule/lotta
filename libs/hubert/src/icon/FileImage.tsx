import * as React from 'react';
import clsx from 'clsx';

import styles from './SvgIcon.module.scss';

export const FileImage = React.memo(
  ({ className, ...props }: React.SVGProps<SVGSVGElement>) => (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      focusable="false"
      viewBox="0 0 384 512"
      aria-hidden="true"
      className={clsx(styles.root, className)}
      data-testid="file-image-icon"
      {...props}
    >
      <path
        fill="#FFCC00"
        d="M64,0C28.7,0,0,28.7,0,64v384c0,35.3,28.7,64,64,64h256c35.3,0,64-28.7,64-64V160H256c-17.7,0-32-14.3-32-32V0
	H64z M256,0v128h128L256,0z M64,256c0-17.7,14.3-32,32-32s32,14.3,32,32s-14.3,32-32,32S64,273.7,64,256z M216,288
	c5.3,0,10.2,2.6,13.2,6.9l88,128c3.4,4.9,3.7,11.3,1,16.5S310,448,304,448h-88h-40h-48H80c-5.8,0-11.1-3.1-13.9-8.1
	s-2.8-11.2,0.2-16.1l48-80c2.9-4.8,8.1-7.8,13.7-7.8s10.8,2.9,13.7,7.8l12.8,21.4l48.3-70.2c3-4.3,7.9-6.9,13.2-6.9V288z"
      />
    </svg>
  )
);
FileImage.displayName = 'FileImageIcon';
