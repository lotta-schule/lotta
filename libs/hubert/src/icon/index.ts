export * from './ArrowUpwardRounded';
export * from './Check';
export * from './ChevronRight';
export * from './Close';
export * from './CloudUpload';
export * from './Copy';
export * from './CreateNewFolder';
export * from './Delete';
export * from './DeleteOutline';
export * from './Download';
export * from './DragHandle';
export * from './Edit';
export * from './ExpandMore';
export * from './File';
export * from './FileArchive';
export * from './FileAudio';
export * from './FileImage';
export * from './FilePDF';
export * from './FilePowerpoint';
export * from './FileTable';
export * from './FileText';
export * from './FileVideo';
export * from './Folder';
export * from './FolderOpen';
export * from './Home';
export * from './KeyboardArrowLeft';
export * from './KeyboardArrowRight';
export * from './MoreVert';
export * from './MoveArrow';
export * from './OpenWith';
