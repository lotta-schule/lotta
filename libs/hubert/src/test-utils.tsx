import * as React from 'react';
import { DefaultThemes } from '@lotta-schule/theme';
import { render, type RenderOptions } from '@testing-library/react';
import { MotionConfig } from 'framer-motion';
import { HubertProvider } from './HubertProvider';
import { GlobalStyles } from './theme';
import { BrowserNode, BrowserState, NodeList } from 'browser';
import { BrowserStateContext } from 'browser/BrowserStateContext';
import * as fixtures from 'test-fixtures';

const theme = DefaultThemes.standard;

const Wrapper = ({ children }: any) => (
  <HubertProvider>
    <MotionConfig transition={{ duration: 0 }}>
      <GlobalStyles theme={theme} />
      {children}
    </MotionConfig>
  </HubertProvider>
);

const customRender = (
  ui: React.ReactElement<any>,
  renderOptions: RenderOptions = {}
): ReturnType<typeof render> =>
  render(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });

// re-export everything
export * from '@testing-library/react';

// override render method
export { customRender as render };

export const createPromise = <T = void,>() => {
  let resolve: (value: T) => void;
  let reject: (error?: any) => void;
  const promise = new Promise<T>((res, rej) => {
    resolve = res;
    reject = rej;
  });
  return { promise, resolve: resolve!, reject: reject! };
};

// For Testing components that use the BrowserStateContext
export type TestBrowserWrapperProps = {
  children?: React.ReactNode;
  defaultNodes?: BrowserNode[];
} & Partial<Omit<BrowserState, 'onRequestChildNodes' | 'renderNodeList'>>;
export const TestBrowserWrapper = ({
  children,
  mode = 'view-and-edit',
  defaultNodes = fixtures.browserNodes,
  moveNode = vi.fn(),
  renameNode = vi.fn(),
  deleteNode = vi.fn(),
  searchNodes = vi.fn(),
  currentPath = [],
  selected = [],
  isNodeDisabled = vi.fn(() => false),
  onSelect = vi.fn(),
  onNavigate = vi.fn(),
  setCurrentAction = vi.fn(),
  currentAction = null,
  currentSearchResults = null,
  setCurrentSearchResults = vi.fn(),
  canEdit,
  uploadClient = {
    addFile: vi.fn(),
    byState: { uploading: [], pending: [], done: [], error: [] },
    hasErrors: false,
    isSuccess: false,
    isUploading: false,
    currentProgress: null,
    currentUploads: [],
  },
}: TestBrowserWrapperProps) => {
  const [nodes, setNodes] = React.useState<BrowserNode[]>(defaultNodes);
  const [isFilePreviewVisible, setIsFilePreviewVisible] = React.useState(false);

  const onRequestChildNodes = (node: BrowserNode<'directory'> | null) =>
    nodes.filter((n) => n.parent === (node?.id ?? null));

  return (
    <BrowserStateContext.Provider
      value={{
        uploadClient,
        canEdit,
        onRequestChildNodes: async (node) =>
          onRequestChildNodes(node as BrowserNode<'directory'>),
        renderNodeList: ({ path }) => (
          <NodeList path={path} nodes={onRequestChildNodes(path.at(-1)!)} />
        ),
        currentAction,
        onSelect,
        onNavigate,
        currentSearchResults,
        currentPath,
        selected,
        mode,
        isNodeDisabled,
        setCurrentAction,
        setCurrentSearchResults,
        moveNode,
        renameNode,
        deleteNode,
        searchNodes,
        isFilePreviewVisible,
        setIsFilePreviewVisible,
        createDirectory: async (parentNode, name) => {
          setNodes((nodes) => {
            return [
              ...nodes,
              {
                id: `${nodes.length + 1}`,
                name,
                type: 'directory',
                parent: parentNode?.id ?? null,
                meta: {},
              },
            ];
          });
        },
      }}
    >
      {children}
    </BrowserStateContext.Provider>
  );
};

export * as fixtures from './test-fixtures';
