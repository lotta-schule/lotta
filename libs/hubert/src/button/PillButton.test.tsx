import { render } from 'test-utils';
import { PillButton } from './PillButton';
import { Close } from '../icon';

describe('PillButton', () => {
  it('renders correctly', () => {
    const { container } = render(
      <PillButton icon={<Close />}>Button</PillButton>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
