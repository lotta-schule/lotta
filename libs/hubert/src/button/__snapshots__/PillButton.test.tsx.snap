// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PillButton > renders correctly 1`] = `
<div
  data-overlay-container="true"
>
  <style
    data-hubert-css-variables-def="true"
  >
    :root {
    --lotta-primary-color: 255, 87, 34;
    --lotta-navigation-background-color: 51, 51, 51;
    --lotta-navigation-contrast-text-color: 234, 234, 234;
    --lotta-error-color: 255, 0, 0;
    --lotta-success-color: 10, 82, 37;
    --lotta-disabled-color: 97, 97, 97;
    --lotta-text-color: 33, 33, 33;
    --lotta-label-text-color: 158, 158, 158;
    --lotta-primary-contrast-text-color: 255, 255, 255;
    --lotta-box-background-color: 255, 255, 255;
    --lotta-page-background-color: 202, 205, 215;
    --lotta-divider-color: 224, 224, 224;
    --lotta-highlight-color: 224, 224, 224;
    --lotta-banner-background-color: 54, 123, 240;
    --lotta-accent-grey-color: 227, 227, 227;
    --lotta-spacing: 8px;
    --lotta-border-radius: 4px;
    --lotta-text-font-family: Muli;
    --lotta-title-font-family: 'Schoolbell', cursive;
}
  </style>
  <style
    data-hubert-global-styles="true"
  >
    
            * {
                box-sizing: border-box;
            }

            html,
            body {
                margin: 0;
                font-family: var(--lotta-text-font-family);
            }

            body {
                margin: 0 auto;

            }
            body::after {
                content: "";
                position: fixed;
                top: 0;
                height: 100vh; /* fix for mobile browser address bar appearing disappearing */
                left: 0;
                right: 0;
                z-index: -1;
                background-color: rgba(var(--lotta-page-background-color), 1);
                background-attachment: scroll;
                background-size: cover;
            }

            ul {
                padding: 0;
                margin: 0;
                list-style: none;
            }

            a {
                color: inherit;
                text-decoration: inherit;
            }

            figure {
                margin: 0;
            }

            h1,
            h2,
            h3,
            h4,
            h5,
            h6 {
                margin-block-start: 0;
                margin-block-end: calc(0.5 * var(--lotta-spacing));
                font-family: var(--lotta-title-font-family);
            }

            h1 {
                font-size: 2em;
            }
            h2 {
                font-size: 1.5em;
            }
            h3 {
                font-size: 1.25em;
            }
            h4 {
                font-size: 1.1em;
            }
            h5,
            h6 {
                font-size: 1em;
            }

            fieldset {
                border: none;
                padding: 0;
            }

            @media print {
              html, body {
                font-size: 8pt;
              }
            }
        
  </style>
  <button
    aria-current="false"
    class="_root_0977c5 _root_3de4f3 _root_6e7a63 _button_6e7a63 _onlyIcon_3de4f3"
    data-variant="default"
    role="button"
    type="button"
  >
    <span
      class="_icon_3de4f3"
    >
      <svg
        aria-hidden="true"
        class="_root_2ade80"
        data-testid="close-icon"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
        />
      </svg>
    </span>
    <span
      class="_label_3de4f3"
    >
      Button
    </span>
  </button>
</div>
`;
